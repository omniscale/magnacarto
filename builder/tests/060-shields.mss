Map {
    background-color: #fff;
}

#roads[type='motorway'] {
    line-width: 1;
}

#labels_roads_refs {
  [type='motorway'] {
    shield-name: "[ref]";
    shield-size: 8;
    shield-face-name: "DejaVu Sans Book";
    shield-fill: #eee;
    shield-file: url(img/rail-24.svg);
    shield-avoid-edges: true;
    shield-clip: false;
    shield-placement: line;
    /*shield-allow-overlap: true;*/
    shield-min-distance: 250;
    shield-min-padding: 50;
    shield-spacing: 250;
    [reflen=5]  { shield-file: url(img/rail-24.svg); }
    [reflen>=6] { shield-file: url(img/rail-24.svg); }
  }
}
