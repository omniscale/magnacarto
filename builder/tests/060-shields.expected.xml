<Map srs="epsg:3857" background-color="#ffffff">
  <Parameters></Parameters>
  <FontSet name="fontset-1">
    <Font face-name="DejaVu Sans Book"></Font>
  </FontSet>
  <Style name="roads" filter-mode="first">
    <Rule>
      <Filter>([type] = &#39;motorway&#39;)</Filter>
      <LineSymbolizer stroke-width="1"></LineSymbolizer>
    </Rule>
  </Style>
  <Style name="labels_roads_refs" filter-mode="first">
    <Rule>
      <Filter>(([reflen] &gt;= 6) and ([type] = &#39;motorway&#39;))</Filter>
      <ShieldSymbolizer avoid-edges="true" clip="false" file="img/rail-24.svg" fill="#eeeeee" fontset-name="fontset-1" minimum-distance="250" minimum-padding="50" placement="line" size="8" spacing="250">[ref]</ShieldSymbolizer>
    </Rule>
    <Rule>
      <Filter>(([reflen] = 5) and ([type] = &#39;motorway&#39;))</Filter>
      <ShieldSymbolizer avoid-edges="true" clip="false" file="img/rail-24.svg" fill="#eeeeee" fontset-name="fontset-1" minimum-distance="250" minimum-padding="50" placement="line" size="8" spacing="250">[ref]</ShieldSymbolizer>
    </Rule>
    <Rule>
      <Filter>([type] = &#39;motorway&#39;)</Filter>
      <ShieldSymbolizer avoid-edges="true" clip="false" file="img/rail-24.svg" fill="#eeeeee" fontset-name="fontset-1" minimum-distance="250" minimum-padding="50" placement="line" size="8" spacing="250">[ref]</ShieldSymbolizer>
    </Rule>
  </Style>
  <Layer name="roads" srs="" status="off">
    <StyleName>roads</StyleName>
  </Layer>
  <Layer name="labels_roads_refs" srs="" status="off">
    <StyleName>labels_roads_refs</StyleName>
  </Layer>
</Map>